lib/beaker-pe/install/pe_utils.rb

```diff
@@ -448,6 +455,35 @@ def deploy_frictionless_to_master(host)
           end
         end
 
+        # Check for availability of required network resources
+        # @param [Array<Host>] hosts
+        # @param [Array<String>] network_resources, defaults to NETWORK_RESOURCES
+        #
+        # @example
+        #   verify_network_resources(hosts, network_resources)
+        #
+        # @return nil
+        #
+        # @api private
+        #
+        def verify_network_resources(hosts, network_resources = NETWORK_RESOURCES)
+          logger.notify("Checking the availability of network resources.")
+          hosts.each do |host|
+            network_resources.each do |resource|
+              on host, "curl -s #{resource} > /dev/null", :acceptable_exit_codes => (0...256)
+              if host.connection.logger.last_result.exit_code != 0
+                logger.warn("Connection error: #{host.host_hash[:vmhostname]} was unable to connect to #{resource}. Please ensure that your test does not require this resource.")
+              end
+            end
+            hosts.each do |target_host|
+              on host, "ping -c1 #{target_host.host_hash[:vmhostname]} > /dev/null", :acceptable_exit_codes => (0...256)
```

Comment:
Does this command work on all platforms?

`-c` doesn't seem to be a valid windows flag: https://technet.microsoft.com/en-us/library/cc940091.aspx

Reply:
@mwbutcher Yeah you probably want `-n 1` here on windows.

Reply:
Also, wouldn't we want the test to fail if it cannot ping the resources? So `0` should be the only valid exit code?

Reply:
Or maybe build up an array of all non-zero exit codes and fail the dest if there are any? Seems silly to do this check and then proceed with testing the node if we can't reach critical resources.

Reply:
There needs to be a way to disable this, or partition the list of addresses - remember this is a public gem that people outside the company use. For released versions, customers don't need (and won't have access) too:


```
"enterprise.delivery.puppetlabs.net",
 "module-staging.puppet.com",
 "forge-aio01-petest.puppetlabs.com",
```

Reply:
Yes, one option might be that instead of having a constant array of resources to test, that the test suite (such as pe_acceptance_tests) provide that list.  It could insert into the options or into each host.  That way it does nothing outside of puppet, and each test suite can decide what resources it considers important.

Reply:
As far as the test failing if we can't find the resource; I'm not certain.  This is meant to be diagnostic, so I kind of feel that it's more important to get the information and any later test failures as well for comparison.